src/state/reducer.ts

```diff
@@ -13,6 +13,13 @@ type Writable<T> = {
   [P in keyof T]: NonNullable<T[P]>;
 }
 
+function resourceCustomizer(a: any, b: any): any {
```

Comment:
can you make this generic so return type is same as input params?

Reply:
Making this typed in any way will be misleading, as the inputs really can be anything. I'll change it to be explicitly typed as a `MergeWithCustomizer` from lodash.

Reply:
that seems to be breaking the tests as it's not found in import?

Reply:
Apperantly thats caused by https://github.com/lodash/babel-plugin-lodash/issues/180

Reply:
I looked at @types/lodash and MergeWithCustomizer isn't exported. Typescript is structurally not nominally typed tho, so you don't need to use the type they provide - it simply has to match the structure.

```typescript
    // mergeWith

    type MergeWithCustomizer = { bivariantHack(value: any, srcValue: any, key: string, object: any, source: any): any; }["bivariantHack"];
```